src/System.Diagnostics.PerformanceCounter/src/misc/EnvironmentHelpers.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Security;
+using System.Security.Principal;
+
+namespace System
+{
+    internal static class EnvironmentHelpers
+    {
+        private static volatile bool s_isAppContainerProcess;
+        private static volatile bool s_isAppContainerProcessInitalized;
+        internal const int TokenIsAppContainer = 29;
+
+        public static bool IsAppContainerProcess
+        {
+            get
+            {
+                if(!s_isAppContainerProcessInitalized)
+                {
+                   if(Environment.OSVersion.Platform != PlatformID.Win32NT)
```

Comment:
I expect that only one path is still hit here in .NET Core. I would debug that code and remove other branches that are obsolete now.

Reply:
I had to add these back after some test failures.

Reply:
On Windows (which is the only relevant platform here) the value is always Win32NT, as @ViktorHofer suggests:

https://github.com/dotnet/corefx/blob/aaaffdf7b8330846f6832f43700fbcc060460c9f/src/System.Runtime.Extensions/src/System/Environment.Windows.cs#L154

You can remove the other codepaths. If there is a test issue, I would investigate that.

Reply:
I mean you can remove the code path that checks for not being Win32NT. You still need the check for whether it's Windows 7 before calling HasAppContainerToken().